crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -489,3 +531,28 @@ macro_rules! impl_system_function {
 }
 
 all_tuples!(impl_system_function, 0, 16, F);
+
+/// Used to implicitly convert systems to their default labels. For example, it will convert
+/// "system functions" to their [`SystemTypeIdLabel`].
+pub trait AsSystemLabel<Marker> {
+    type SystemLabel: SystemLabel;
+    fn as_system_label(&self) -> Self::SystemLabel;
```

Comment:
I'll record that I'm still not super happy with this being a method, but I don't think that's changing.

Reply:
Yeah I'm not (yet) convinced. I think there are good reasons to make "manually retrieving the label from a system instance" a part of our api. We don't abstract out "other" labels and let those instances float around. Why abstract out these?
I'm down to continue discussing, but I dont want to block on this.